src/runtime/manifest-string-builder.ts

```diff
@@ -0,0 +1,67 @@
+/**
+ * @license
+ * Copyright (c) 2020 Google Inc. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * Code distributed by Google as part of this project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+const spacesPerIndent = 2;
+
+export class ManifestStringBuilder {
+  private readonly lines: string[];
+  private readonly indent: number;
+  private readonly startIndex: number;
+
+  constructor(parentBuilder?: ManifestStringBuilder) {
+    if (parentBuilder) {
+      this.lines = parentBuilder.lines;
+      this.indent = parentBuilder.indent + 1;
+      this.startIndex = this.lines.length;
+    } else {
+      this.lines = [];
+      this.indent = 0;
+      this.startIndex = 0;
+    }
+  }
+
+  /** Add the given line(s) with the correct indent level. */
+  push(...lines: string[]) {
+    this.lines.push(...lines.map(line => ' '.repeat(this.indent * spacesPerIndent) + line));
```

Comment:
optional - do we want to do `line.trim()` to avoid the caller sending a string with indent by mistake? not sure if this is a good idea.

Reply:
That might cause accidents. I think I'd rather it failed straight away so the code can be fixed up instead.